### PR TITLE
Wrapping offlineStorage initialization inside a function

### PR DESCRIPTION
**Pull request by cahirodoherty-learningpool**
This is to prevent loading on import. The ready state was being set to false, but depending on extension configuration may not be set to ready. ie LanguagePicker was preventing app:dataLoaded call from firing so user was faced with persistent loading screen

Deals with issue: https://github.com/adaptlearning/adapt-contrib-xapi/issues/85

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -2,17 +2,19 @@ define([
   'core/js/adapt',
   './adapt-contrib-xapi',
   './adapt-offlineStorage-xapi'
-], function(Adapt) {
+], function(Adapt, xapi, offlineStorage) {
 
   Adapt.on('app:dataLoaded', initialise);
```
Comment (oliverfoster) [inline js/index.js:7]: offlineStorage:prepare might be a better event, if you look at the ConfigModel you'll see there's a special event for initializing the offlineStorage handler.

Comment (cahirodoherty-learningpool) [reply js/index.js]: Triggering on this actually leaves the initialization hanging due to the order of events triggered I believe

Comment (oliverfoster) [reply js/index.js:7]: Weird. Good to know though. Wish I had more time to take a proper look.

File: js/index.js [hunk 2]
```diff
@@ -22,9 +24,10 @@ define([
     Adapt.once('xapi:lrs:initialize:error ' + successEvent, function() {
       Adapt.offlineStorage.get();
       Adapt.offlineStorage.setReadyStatus();
-      Adapt.wait.end();
     });
 
+    Adapt.wait.end();
```
Comment (oliverfoster) [inline js/index.js:29]: I'm not sure you need these waits at all now as the code is synchronous.

Comment (cahirodoherty-learningpool) [reply js/index.js:29]: I was just thinking this last night. I'll test out those changes and submit it if they are all good 👍

Comment (cahirodoherty-learningpool) [reply js/index.js]: Unfortunately offlineStorage:prepare isn't firing so I have had to stick with the app:dataLoaded as before

**Pull request merged**